src/moonlink/src/storage/mooncake_table.rs

```diff
@@ -656,6 +684,35 @@ impl MooncakeTable {
         })
     }
 
+    pub(crate) fn alter_table(
+        &mut self,
+        alter_table_request: AlterTableRequest,
+    ) -> Arc<TableMetadata> {
+        assert!(
+            self.mem_slice.get_num_rows() == 0,
+            "Cannot alter table with non-empty mem slice"
+        );
+
+        let new_metadata = Arc::new(TableMetadata::new_for_alter_table(
+            self.metadata.clone(),
+            alter_table_request,
+        ));
+
+        let mut guard = self.snapshot.try_write().unwrap();
+        guard.reset_for_alter(new_metadata.clone());
+        assert!(
+            self.metadata.schema.fields.len() != new_metadata.schema.fields.len(),
+            "Only support alter table with add/drop fields"
+        );
+        self.mem_slice = MemSlice::new(
```

Comment:
I did something similar in the latest commit: https://github.com/Mooncake-Labs/moonlink/blob/317a13442cc8effd3159ba284ac4c32afb49b270/src/moonlink/src/storage/mooncake_table.rs#L1085-L1096
Do you mind rebasing?